main.cpp

```diff
@@ -9,7 +9,9 @@ int main()
     std::cout << std::endl << "OPEL" << std::endl;
     PetrolCar opel(new PetrolEngine(120, 1800, 6));
     car = &opel;
+    car->changeGear(4);
     car->accelerate(50);
+    car->changeGear(-1);
```

Comment:
I suggest making ```enum class``` for gear which will be more self descriptive than "raw" integer values.

e.g.
```
enum class FiveSpeedGear
{
	_R = 0,
	_1 = 1,
	_2,
	_3,
	_4,
	_5
};

//and then:

//...
FiveSpeedGear gear = FiveSpeedGear::_1;
//...
```
see: https://ideone.com/f2rXsg

It has a lot of advantages, like:
* prevents you from using value out of the scope defined in enum (e.g. any integer value in this case)
* is a lot more descriptive and easier to use self-descriptive enum type rather than integer, which can mean anything
* enum class is better than plain enum because it can't be implicitly casted to any other type (some int in most of the cases or even another enum!)
* enum class is scoped, rather than plain enum, which is not (see: https://ideone.com/xdjBHX and https://ideone.com/GYeavX)

Reply:
Okay, i get it. That's a good point.
Changes will be made as soon as possible.